packages/clerk-js/src/ui/components/SignIn/SignInStart.tsx

```diff
@@ -64,42 +64,73 @@ export function _SignInStart(): JSX.Element {
     placeholder: localizationKeys('formFieldInputPlaceholder__password') as any,
   });
 
-  const identifierField = useFormControl('identifier', '', {
+  const ctxInitialValues = ctx.initialValues || {};
+  const initialValues: Record<SignInStartIdentifier, string | undefined> = useMemo(
+    () => ({
+      email_address: ctxInitialValues.emailAddress,
+      email_address_username: ctxInitialValues.emailAddress || ctxInitialValues.username,
+      username: ctxInitialValues.username,
+      phone_number: ctxInitialValues.phoneNumber,
+    }),
+    [ctx.initialValues],
+  );
+
+  const hasSocialOrWeb3Buttons = !!authenticatableSocialStrategies.length || !!web3FirstFactors.length;
+  const [shouldAutofocus, setShouldAutofocus] = useState(!isMobileDevice() && !hasSocialOrWeb3Buttons);
+  const textIdentifierField = useFormControl('identifier', initialValues[identifierAttribute] || '', {
     ...currentIdentifier,
     isRequired: true,
   });
 
+  const phoneIdentifierField = useFormControl('identifier', initialValues['phone_number'] || '', {
+    ...currentIdentifier,
+    isRequired: true,
+  });
+
+  const identifierField = identifierAttribute === 'phone_number' ? phoneIdentifierField : textIdentifierField;
+
+  const identifierFieldRef = useRef<HTMLInputElement>(null);
+
   const switchToNextIdentifier = () => {
     setIdentifierAttribute(
       i => identifierAttributes[(identifierAttributes.indexOf(i) + 1) % identifierAttributes.length],
     );
-    identifierField.setValue('');
+    setShouldAutofocus(true);
   };
 
-  const switchToPhoneInput = (value?: string) => {
+  const switchToPhoneInput = () => {
     setIdentifierAttribute('phone_number');
-    identifierField.setValue(value || '');
+    setShouldAutofocus(true);
```

Comment:
❓ Does this autofocus resolve a UI issue that was discovered as part of this work?